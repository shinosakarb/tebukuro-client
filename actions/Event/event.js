// @flow
import { createAction } from 'redux-actions'
import Router from 'next/router'
import type { Dispatch } from 'redux'

import ActionsType from '../../constants/Actions'
import type { EventProps } from '../../types/Event'
// TODO: Enable ESLint and flow check after API implemented.
/* eslint-disable */
// flow-disable-nextline
import EventAPI from '../../api/Event'

const create = createAction(ActionsType.Event.createEvent)

export const createEvent = (params: EventProps) => (dispatch: Dispatch) => (
  EventAPI.create(params)
    .then((json) => {
      Router.replace(`/event/${json.id}`)
      return dispatch(create(json))
    })
    .catch(error => dispatch(create(error)))
)
