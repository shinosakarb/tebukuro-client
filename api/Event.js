// @flow
import pathToRegexp from 'path-to-regexp'
import moment from 'moment-timezone'
import Base from './Base'

type createProps = {
  name: string,
  description: string,
  quota: number,
  eventStartsAt: string,
  eventEndsAt: string,
}

const toISOString = (datetime: string) => moment(datetime).format()

export default class Event extends Base {
  cancelRegistration = (params: number) => {
    const url = pathToRegexp.compile(this.endpoints.cancelRegistration)(params)
    return this.client.delete(url).then(this.onSuccess, this.onFailure)
  }

  createEvent = (params: createProps) => {
    const createParams = {
      ...params,
      eventStartsAt: toISOString(params.eventStartsAt),
      eventEndsAt: toISOString(params.eventEndsAt),
    }

    return this.create(createParams)
  }
}
