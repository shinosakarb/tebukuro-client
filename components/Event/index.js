import PropTypes from 'prop-types'

const Event = props => (
  <div>
    <p>name: {props.event.name}</p>
    <p>description: {props.event.description}</p>
  </div>
)

Event.propTypes = { event: PropTypes.instanceOf(Object).isRequired }

export default Event
