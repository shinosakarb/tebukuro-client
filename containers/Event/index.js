import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import EventComponent from '../../components/Event'

export const Event = props => (
  <div>
    This is the Event container.
    <EventComponent event={props.event} />
  </div>
)

Event.propTypes = { event: PropTypes.instanceOf(Object).isRequired }

export default connect(null)(Event)
