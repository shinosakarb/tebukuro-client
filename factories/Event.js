export default {
  event1: {
    id: 1,
    name: 'event1',
    description: 'This is the first event.',
  },
}
