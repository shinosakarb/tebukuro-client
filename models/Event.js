// @flow
import { Set } from 'immutable'
import Record from './Base'

export const EventRecord = Record({
  id: 0,
  name: '',
  description: '',
  participants: Set.of(),
})

export default class Event extends EventRecord {
  setParticipants(list: ?Object[]) {
    return this.set('participants', Set.of(...list))
  }
}
