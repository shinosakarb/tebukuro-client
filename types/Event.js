// @flow

export type EventId = {
  id: number
}

export type EventProps = {
  id: ?number,
  name: string,
  description: string,
  quota: number,
  eventStartsAt: string,
  eventEndsAt: string,
  registered: boolean,
  participants: Object[],
  errors: ?string[]
}
