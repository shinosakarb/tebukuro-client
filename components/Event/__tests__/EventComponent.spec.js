import { shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import Event from '../index'
import EventParams from '../../../factories/Event'

const testParams = {
  ...EventParams.event1,
  participants: [1, 2, 3],
}

describe('Event Component', () => {
  it('renders self.', () => {
    const component = shallow(<Event event={testParams} />)
    const tree = shallowToJson(component)

    expect(tree).toMatchSnapshot()
  })
})
