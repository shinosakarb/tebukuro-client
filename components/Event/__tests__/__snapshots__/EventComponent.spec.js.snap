// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component renders self. 1`] = `
<div>
  <p>
    name: 
    event1
  </p>
  <p>
    description: 
    This is the first event.
  </p>
  <p>
    quota: 
    10
  </p>
  <p>
    number of participants: 
    3
  </p>
</div>
`;
